app/api/submission.py

```diff
@@ -9,68 +12,86 @@
 from app.utils.img_processing.google_api import GoogleAPI, NoTextFoundException
 from app.api.models import Submission, ImageSubmission
 
-# global variables and services
+# Allows user to set environment variable to use different Tesseract model
+OCR_MODEL = os.getenv("OCR_MODEL", default="storysquad")
+
+# Global variables and services
 router = APIRouter()
 log = logging.getLogger(__name__)
-vision = GoogleAPI()
+# Google Vision is used for drawings so is needed regardless if Tesseract is used for OCR
+google_vision = GoogleAPI() 
 
 
 @router.post("/submission/text")
-async def submission_text(sub: Submission):
-    """Takes a Submission Object and calls the Google Vision API to text annotate
+def submission_text(sub: Submission):
+    """
+    Takes a Submission Object and calls the API (Google or Tesseract) to text annotate
     the passed s3 link, then passes those concatenated transcriptions to the SquadScore
     method, returns:
 
     Arguments:
     ---
     `sub`: Submission - Submission object **see `help(Submission)` for more info**
+    
     Returns:
     ---
     ```
-    {"SubmissionID": int, "IsFlagged": boolean,"LowConfidence": boolean, "Complexity": int}
+    {"SubmissionID": int, "IsFlagged": boolean, "LowConfidence": boolean, "Complexity": int}
     ```
     """
+    # Set model for OCR, defaults to Google Vision
+    if sub.Model == "tesseract":
+        vision = TesseractAPI(OCR_MODEL)
```

Comment:
you should probably set a `tesseract_vision = TesseractAPI(OCR_MODEL)` on the global level on line 23 after where its done for goog.e The way this is written, every time a submission is added, the API will have to load the tesseract model from disk which is extra overhead. If the model that ends up being used is very large, this can be very expensive.

 It also means that if there are simultaneous requests, the model will be loaded more than once. This means its possible to execute a denial of service attack by hitting the endpoint enough that the machine runs out of memory from all the models getting loaded into memory.

Reply:
That makes perfect sense thanks @nlittlepoole I've gone ahead and done as you suggested creating a new tesseract_vision variable on the global level and using the if statement to determine which is used.